components/script/script_thread.rs

```diff
@@ -1761,6 +1758,13 @@ impl ScriptThread {
         ROUTER.route_ipc_receiver_to_mpsc_sender(ipc_timer_event_port,
                                                  self.timer_event_chan.clone());
 
+        // If this is an about:blank load, it must share the creator's origin.
```

Comment:
Hmm fair, I also noticed that this only needs to be done in when handling attach layout as that will pick the proper origin and store it in the in progress load struct.